src/tokens/aoTokens/ao.ts

```diff
@@ -144,7 +145,7 @@ export function useAoTokens(): [TokenInfoWithBalance[], boolean] {
               const balance = Number(
                 await timeoutPromise(
                   (async () => {
-                    const aoToken = await Token(id);
+                    const aoToken = await Token(id, null, ao);
```

Comment:
@7i7o Should we not use `Token` here and directly fetch balance as this first fetches the token info and then the balance ? Can we directly fetch the balance and use the `Denomination` saved on the storage ?

Reply:
We could, but when could we update the Denomination if someone changes that?

Reply:
Yes, that can be the issue if we use the stored one and if Denomination of a token changes. We can leave it as it is.